frontend/src/metabase/query_builder/components/DatasetEditor/DatasetEditor.jsx

```diff
@@ -433,7 +434,7 @@ function DatasetEditor(props) {
                 id: "metadata",
                 name: t`Metadata`,
                 icon: "label",
-                disabled: !hasResultsMetadata,
+                disabled: resultsMetadata === null,
```

Comment:
seems like this could be `undefined` as well since it is coming directly from the selector. maybe we should just use `==`?